src/quantum/azext_quantum/operations/workspace.py

```diff
@@ -0,0 +1,205 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+# pylint: disable=line-too-long,redefined-builtin
+
+from knack.util import CLIError
+
+from .._client_factory import cf_workspaces, cf_quotas
+from ..vendored_sdks.azure_mgmt_quantum.models import QuantumWorkspace
+from ..vendored_sdks.azure_mgmt_quantum.models import QuantumWorkspaceIdentity
+from ..vendored_sdks.azure_mgmt_quantum.models import Provider
+from msrestazure.azure_exceptions import CloudError
+
+import time
+
+DEFAULT_WORKSPACE_LOCATION = 'westus'
+POLLING_TIME_DURATION = 3
+MAX_RETRIES_ROLE_ASSIGNMENT = 10
+
+
+class WorkspaceInfo(object):
+    def __init__(self, cmd, resource_group_name=None, workspace_name=None, location=None):
+        from azure.cli.core.commands.client_factory import get_subscription_id
+
+        # Hierarchically selects the value for the given key.
+        # First tries the value provided as argument, as that represents the value from the command line
+        # then it checks if the key exists in the 'quantum' section in config, and uses that if available.
+        # finally, it checks in the 'global' section in the config.
+        def select_value(key, value):
+            if value is not None:
+                return value
+            value = cmd.cli_ctx.config.get('quantum', key, None)
+            if value is not None:
+                return value
+            value = cmd.cli_ctx.config.get(cmd.cli_ctx.config.defaults_section_name, key, None)
+            return value
+
+        self.subscription = get_subscription_id(cmd.cli_ctx)
+        self.resource_group = select_value('group', resource_group_name)
+        self.name = select_value('workspace', workspace_name)
+        self.location = select_value('location', location)
+
+    def clear(self):
+        self.subscription = ''
+        self.resource_group = ''
+        self.name = ''
+        self.location = ''
+
+    def save(self, cmd):
+        from azure.cli.core.util import ConfiguredDefaultSetter
+
+        with ConfiguredDefaultSetter(cmd.cli_ctx.config, False):
+            cmd.cli_ctx.config.set_value('quantum', 'group', self.resource_group)
+            cmd.cli_ctx.config.set_value('quantum', 'workspace', self.name)
+            cmd.cli_ctx.config.set_value('quantum', 'location', self.location)
+
+
+def _show_tip(msg):
+    import colorama
+    colorama.init()
+    print(f"\033[1m{colorama.Fore.YELLOW}{msg}{colorama.Style.RESET_ALL}")
```

Comment:
Usually, we output warnings to `stderr`. Is there any reason to allow them to be sent to `stdout`?